client/js/verification-portal/components/header-modal.js

```diff
@@ -44,40 +47,34 @@ class HeaderModal extends Component {
 			this.state.closed && classes.closed,
 		)
 
-		if (
-			(submitStatus === 'Success' && this.state.prevSubmit === null) ||
-			this.state.prevSubmit === 'Success'
-		) {
-			return (
-				<div className={modalClasses}>
-					<div>
-						<p className={classes.title}>Information Submitted Successfully</p>
-						<p className={classes.caption}>
-							{
-								"Thanks for helping us bring better healthcare to women's fingertips!"
-							}
-						</p>
-					</div>
-					<div className={classes.button} onClick={this.handleClick}>
-						&times;
-					</div>
+		let title
+		let caption
+		if (submitStatus === 'Success') {
+			title = 'Information Submitted Successfully'
+			caption =
+				"Thanks for helping us bring better healthcare to women's fingertips!"
+		} else if (submitStatus === 'Failed') {
+			title = 'Whoops!'
+			caption = 'An error occurred and the HAH staff has been notified.'
+		}
+
+		return (
+			<div className={modalClasses}>
+				<div>
+					<p className={classes.title}>{title}</p>
+					<p className={classes.caption}>{caption}</p>
 				</div>
-			)
-		} else {
-			return (
-				<div className={modalClasses}>
-					<div>
-						<p className={classes.title}>Whoops!</p>
-						<p className={classes.caption}>
-							An error occurred and the HAH staff has been notified.
-						</p>
-					</div>
-					<div className={classes.button} onClick={this.handleClick}>
-						&times;
-					</div>
+				<div
+					className={classes.button}
+					onClick={() => {
+						store.dispatch(setFormStatus('Pending'))
```

Comment:
I added this line. If the 'x' is clicked on the modal, the form status is set back to pending so that when the Save button is clicked again, the appropriate modal is shown. Without doing this, if you get the "success" modal 2x in a row, it won't show the 2nd time because the state of the form status hasn't changed (meaning a refresh of the component isn't triggered).